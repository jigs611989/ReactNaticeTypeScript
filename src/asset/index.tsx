export { default as Color } from './Color';
export { default as AppStyle } from './AppStyle';
export { default as String } from './String';
export { default as Images } from './image';
