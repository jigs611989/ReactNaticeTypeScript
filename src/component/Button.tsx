import React from 'react';
import { Text, TouchableOpacity } from 'react-native';
import styles from './style/Button';

type Props = {
  title: string,
  onPress: () => void,
  style?: object,
  textStyle?: object,
}

const Button: React.FC<Props> = ({title, onPress, style, textStyle}) => {
  return (
    <TouchableOpacity onPress={onPress} style={[styles.button, style]}>
      <Text style={[styles.text, textStyle]}>{title}</Text>
    </TouchableOpacity>
  )
}

export default Button;
