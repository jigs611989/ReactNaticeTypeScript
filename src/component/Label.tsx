import React from 'react';
import { Text } from 'react-native';
import styles from './style/Label';

type Props = {
  title: string;
  style?: object;
};

const Label: React.FC<Props> = ({ style, title, ...props }) => (
  <Text 
    style={[styles.title, style]}
    {...props}
  >
    {title}
  </Text>
);

export default Label;
