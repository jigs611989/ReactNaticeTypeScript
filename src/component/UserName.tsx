import React from 'react';
import { View, Text } from 'react-native';
import { useSelector } from 'react-redux';
import style from './style/UserName';

const UserName = () => {
  const name = useSelector((state) => state.name);

  return (
    <View style={style.container}>
      <Text style={style.text}>{name}</Text>
    </View>
  );
};

export default UserName;
