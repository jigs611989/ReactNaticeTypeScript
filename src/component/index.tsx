export { default as Button } from './Button';
export { default as Label } from './Label';
export { default as UserName } from './UserName';
export { default as TextField } from './TextField';
export { default as SliderButton } from './SliderButton';
