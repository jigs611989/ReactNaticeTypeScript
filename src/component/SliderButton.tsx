import React from 'react';
import { Image, Text } from 'react-native';
import Slider from 'react-native-slide-to-unlock';
import { String } from '../asset';
import style from './style/SliderButton';

type Props = {
  onPress: () => void,
  title?: string,
  childrenContainer?: object,
  containerStyle?: object,
  sliderElement?: React.Component,
  testID?: string
};

const SliderButton: React.FC<Props> = ({
  onPress,
  title,
  sliderElement,
  childrenContainer,
  containerStyle,
  testID
}) => {
  return (
    <Slider
      childrenContainer={childrenContainer}
      onEndReached={onPress}
      containerStyle={[style.containerStyle, containerStyle]}
      sliderElement={
        sliderElement ? (
          sliderElement
        ) : (
          <Image
            style={style.imageStyle}
            source={{
              uri:
                'https://facebook.github.io/react-native/docs/assets/favicon.png',
            }}
          />
        )
      }>
      <Text>{title ? title : String.Buttons.slideMe}</Text>
    </Slider>
  );
};

export default SliderButton;
