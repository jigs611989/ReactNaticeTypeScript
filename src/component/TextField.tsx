import React from 'react';
import { TextInput, TextInputProps, View } from 'react-native';
import { Color } from '../asset';
import styles from './style/TextField';

type TextFieldProps = TextInputProps & {
  containerStyle?: object;
  inputStyle?: object;
};

const TextField: React.FC<TextFieldProps> = ({
  containerStyle,
  inputStyle,
  ...props
}) => {
  return (
    <View style={[styles.container, containerStyle]}>
      <TextInput
        placeholderTextColor={Color.placeholder}
        style={[styles.input, inputStyle]}
        {...props}
      />
    </View>
  );
};

export default TextField;
