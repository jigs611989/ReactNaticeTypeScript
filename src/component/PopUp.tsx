import React, {
  forwardRef,
  useCallback,
  useImperativeHandle,
  useState,
} from 'react';
import { Modal, View } from 'react-native';
import { BlurView } from '@react-native-community/blur';
import Label from './Label';
import Button from './Button';
import styles from './style/PopUp';

const Backdrop = () => (
  <BlurView style={styles.backdrop} blurType="light" blurAmount={5} />
);

type Props = {
  title: string;
  description?: string;
  onPress: () => void;
};

const PopUp = (props: Props, ref: any) => {
  const [isVisible, setIsVisible] = useState(false);

  const show = useCallback(() => setIsVisible(true), []);
  const hide = useCallback(() => setIsVisible(false), []);

  useImperativeHandle(ref, () => ({
    show,
    hide,
  }));

  const { title, description, onPress } = props;

  return (
    <Modal
      testID="ErrorPopupModal"
      visible={isVisible}
      onRequestClose={() => setIsVisible(false)}
      transparent={true}>
      <View style={styles.container}>
        <Backdrop />
        <View style={styles.popUp}>
          <Label title={title} style={styles.title} />
          {description ? (
            <Label title={description} style={styles.description} />
          ) : null}
          <Button title={'Okay'} onPress={onPress} style={styles.buttonStyle} />
        </View>
      </View>
    </Modal>
  );
};

export default forwardRef(PopUp);
