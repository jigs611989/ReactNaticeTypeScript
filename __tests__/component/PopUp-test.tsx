import React, { createRef } from 'react';
import { PopUp } from '../../src/component';
import { render } from '@testing-library/react-native';

describe('PopUp', () => {
  
  it('should have a title & description', () => {
    
    const popUp = createRef();
    const title = 'popUpTest';
    const descriptionText = 'testing descption';
    const mockOnPress = () => {};
    const { toJSON, getByText } = render(
      <PopUp
        title={title}
        description={descriptionText}
        onPress={mockOnPress}
        ref={popUp}
      />,
    );
    const popupTitle = getByText(title);
    expect(popupTitle).toBeDefined();
    const descriptionLabel = getByText(descriptionText);
    expect(descriptionLabel).toBeDefined();

    // should match the snapshot
    expect(toJSON()).toMatchSnapshot();
  });
});
