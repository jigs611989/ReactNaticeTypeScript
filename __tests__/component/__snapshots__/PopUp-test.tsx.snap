// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopUp should have a title & description 1`] = `
<Modal
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  testID="ErrorPopupModal"
  transparent={true}
  visible={false}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View />
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 12,
          "elevation": 1,
          "justifyContent": "center",
          "maxHeight": 740,
          "maxWidth": 340,
          "paddingHorizontal": 20,
          "paddingVertical": 30,
          "shadowColor": "#0B121E",
          "shadowOffset": Object {
            "height": -5,
            "width": 0,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 15,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 24,
              "lineHeight": 28,
              "margin": 16,
              "marginBottom": 2,
            },
            Object {
              "color": "#0B121E",
              "fontSize": 18,
              "fontWeight": "bold",
              "marginVertical": 9,
              "textAlign": "center",
            },
          ]
        }
      >
        popUpTest
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 24,
              "lineHeight": 28,
              "margin": 16,
              "marginBottom": 2,
            },
            Object {
              "color": "#747577",
              "fontSize": 16,
              "marginBottom": 32,
              "textAlign": "center",
            },
          ]
        }
      >
        testing descption
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#f7a934",
            "borderRadius": 10,
            "height": 50,
            "justifyContent": "center",
            "margin": 20,
            "opacity": 1,
            "width": 300,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "bold",
              },
              undefined,
            ]
          }
        >
          Okay
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
